openwisp_controller/config/api/serializers.py

```diff
@@ -115,6 +118,53 @@ class Meta:
         }
 
 
+class DeviceConfigMixin(object):
+    def _get_config_templates(self, config_data):
+        return [template.pk for template in config_data.pop('templates', [])]
+
+    @transaction.atomic
+    def _create_config(self, device, config_data):
+        config_templates = self._get_config_templates(config_data)
+        try:
+            if not device._has_config():
+                config = Config(device=device, **config_data)
+                config.full_clean()
+                config.save()
+            else:
+                # If the "device group" was specified, then
+                # the config would get automatically created
+                # for the device. Hence, we perform update
+                # operation on config of a new device.
+                config = device.config
+                for key, value in config_data.items():
+                    setattr(config, key, value)
+                config.full_clean()
```

Comment:
These lines look the same as in the `_update_config` method below, therefore we should move these lines to a private method which is shared between both methods, so that if we ever need to put our hands on these lines we will reduce the chances of forgetting to update both parts.

```suggestion
                config = self._prepare_config(device, config_data)
```